server/src/main/java/org/elasticsearch/common/time/JavaDateFormatter.java

```diff
@@ -98,27 +90,27 @@ JavaDateFormatter getRoundupParser() {
     // subclasses override roundUpParser
     JavaDateFormatter(
         String format,
-        DateTimeFormatter printer,
-        BiConsumer<DateTimeFormatterBuilder, DateTimeFormatter> roundupParserConsumer,
-        DateTimeFormatter... parsers
+        DateTimePrinter printer,
+        BiConsumer<DateTimeFormatterBuilder, DateTimeParser> roundupParserConsumer,
+        DateTimeParser... parsers
     ) {
         if (printer == null) {
             throw new IllegalArgumentException("printer may not be null");
         }
+        if (parsers.length == 0) {
+            throw new IllegalArgumentException("parsers need to be specified");
+        }
         this.printer = printer;
         this.format = format;
-        this.parsers = parsersArray(printer, parsers);
-        this.roundupParser = createRoundUpParser(format, roundupParserConsumer, locale(), this.parsers);
+        this.parsers = parsersArray(parsers);
+        this.roundupParsers = createRoundUpParsers(format, roundupParserConsumer, locale(), this.parsers);
     }
 
-    private static DateTimeFormatter[] parsersArray(DateTimeFormatter printer, DateTimeFormatter... parsers) {
-        if (parsers.length == 0) {
```

Comment:
we don't need this?

Reply:
Now the printer & parser are explicitly separate, we don't default to the printer. The new constructor taking a single DateTimeFormatter handles this instead